http/client/request.go

```diff
@@ -45,6 +50,73 @@ func NewRequest(L *lua.LState) int {
 	return 1
 }
 
+// http.filerequest(url, files, params) returns user-data, error
+func NewFileRequest(L *lua.LState) int {
+	url := L.CheckString(1)
+	files := L.CheckTable(2)
+
+	body := new(bytes.Buffer)
+	writer := multipart.NewWriter(body)
+	var writeFile = func(info *lua.LTable, w *multipart.Writer) (err error) {
+		fieldname := info.RawGetString("fieldname")
+		path := info.RawGetString("path")
+		if fieldname == lua.LNil || path == lua.LNil {
+			return errors.New("fieldname or path is nil")
+		}
+		filename := info.RawGetString("filename")
+		if filename == lua.LNil {
+			filename = lua.LString(filepath.Base(path.String()))
+		}
+
+		part, err := writer.CreateFormFile(fieldname.String(), filename.String())
+		if err != nil {
+			return
+		}
+		file, err := os.Open(path.String())
+		if err != nil {
+			return
+		}
+		defer file.Close()
+		_, err = io.Copy(part, file)
+		return
+	}
+
+	if err := writeFile(files, writer); err != nil{
```

Comment:
This seems strange to me - please explain - it _seems_ like when there's an error, you ignore the error and  write the files, but when no errors, you don't. That seems backwards to me, so need the explanation - maybe in a comment.

Reply:
This is only a compatible method. First, try to process a single file. If it fails, try the table of files. Like this:
```lua
local req, err = http.file_request("http://127.0.0.1:1111/upload", {
  fieldname = "file",
  path = "./test/data/test.txt"
})
local req, err = http.file_request("http://127.0.0.1:1111/uploadMultiple", {
  {fieldname = "file", path = "./test/data/test.txt"},
  {fieldname = "file1", path = "./test/data/test1.txt"}
})
```

Reply:
So… is the point that the files parameter could be either an array or a map?

Could you check the diff by checking `files.Len() == 0` means map - or single file, otherwise, it's an array, then propagate errors? This would let errors returned by passing a single file to be reported rather than swallowed up and confused by trying to iterate over a non-array.